pmlc/dialect/pxa/analysis/strides.cc

```diff
@@ -397,6 +398,59 @@ Optional<StrideInfo> computeStrideInfo(pmlc::dialect::pxa::AffineReduceOp op) {
                            op.getMapOperands());
 }
 
+Optional<RelativeAccessPattern> computeRelativeAccess(Block *block,
+                                                      Operation *op) {
+  ArrayRef<int64_t> vecSize = {};
+  Optional<llvm::SmallVector<StrideInfo, 4>> maybeStrides;
+  TypeSwitch<Operation *>(op)
+      .Case<AffineLoadOp>([&](auto op) {
+        maybeStrides =
+            computeStrideInfo(op.getAffineMap(), op.getMapOperands());
+      })
+      .Case<pmlc::dialect::pxa::AffineReduceOp>([&](auto op) {
+        maybeStrides =
+            computeStrideInfo(op.getAffineMap(), op.getMapOperands());
+      })
+      .Case<AffineVectorLoadOp>([&](auto op) {
+        maybeStrides =
+            computeStrideInfo(op.getAffineMap(), op.getMapOperands());
+        vecSize = op.getVectorType().getShape();
+      })
+      .Case<pmlc::dialect::pxa::AffineVectorReduceOp>([&](auto op) {
+        maybeStrides =
+            computeStrideInfo(op.getAffineMap(), op.getMapOperands());
+        vecSize = op.getVectorType().getShape();
+      });
+  if (!maybeStrides) {
+    return llvm::None;
+  }
+  auto &strides = *maybeStrides;
+  RelativeAccessPattern ret;
+  for (size_t i = 0; i < strides.size(); i++) {
```

Comment:
Feels like some documentation is warranted to describe the math being done in this loop.  This may be just "learning curve" for me but I can't make sense of it.